test/e2e-layering/layering_test.go

```diff
@@ -1,7 +1,171 @@
 package e2e_layering_test
 
-import "testing"
+import (
+	"testing"
 
-func TestHelloWorld(t *testing.T) {
-	t.Log("Hello world")
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	ign3types "github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+)
+
+type LayeringSuite struct {
+	suite.Suite
+}
+
+func (suite *LayeringSuite) SetupSuite() {
+	cs := framework.NewClientSet("")
+
+	// don't keep unlabel function or delete pool function since removing a node from a layered pool is...not well defined
+	helpers.LabelAllNodesInPool(suite.T(), cs, "worker", "node-role.kubernetes.io/"+ctrlcommon.ExperimentalLayeringPoolName)
+	helpers.CreateMCP(suite.T(), cs, ctrlcommon.ExperimentalLayeringPoolName)
+
+	// TODO wait for that to rollout and assert that it rebased
```

Comment:
@jkyros are you preserving the behavior of `WaitForPoolComplete`? It checks `mcp.Status.Configuration.Name` and checks that against a rendered config (I guess I could get the latest rendered config for the layered pool?)

Or should I take a completely different approach for checking for layered rollout?

Reply:
I should be. 

Well, so even though I put us in this weird "config via image" vs "config is the image" place, I think that should still work either way (but I want to go check what we do before we call `WaitForPoolComplete()` to get `target`... )

Like `mcp.Spec.Configuration` and `mcp.Status.Configuration` are just ObjectRefs,  and as long as we don't try to assume their type and do machine-config-y things with them without checking what they are (and I don't think we do in there) it should be fine. 

Today `spec.Config.name` is `rendered-layered-9c7128a1d5e0a3d0359e7956db5c9479` but tomorrow, say it will be `sha256:1d400c8b664fe7be00bd0bd71f7da84aba0a817c85a9b8b061e78d04be2a74d0` and we can still compare object names.